### PR TITLE
AWS - VPC module

### PR DESCRIPTION
**Pull request by laura-herrera**
Requirements:
- Number of AZs, default is 1
- Enable Internet access from Public subnets, default is false
Resources:
- Creates a new VPC
- Sets some DHCP options, configurable
- Creates Public subnets as the AZ number
- Creates Private subnets as the AZ number
- Creates a Route Table for the Public subnets
- Creates a Route Table per Private subnet
- Creates an Internet Gateway if requested
- Adds a route on Public routing table to internet via IGW, if requested
- Creates a Security Group for Public which allows traffic for HTTP, HTTPS
NOTE:
- This module doesn't create any NAT gateways, so private subnets do not have internet access in any case, for now
TODO:
- Add Network ACLs

### CODE REVIEW

**Review by patrick-merchant — APPROVED** [review]
LGTM

File: aws/vpc/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
+output "vpc_id" {
```
Comment (patrick-merchant) [inline aws/vpc/outputs.tf:1]: Could we also output the vpc_name variable, for use in other modules? Or would we not typically output variables?

File: aws/vpc/main.tf [hunk 2]
```diff
@@ -0,0 +1,182 @@
+locals {
+  az = ["a", "b", "c", "d"]
```
[the remark this answers is not in the document]
Comment (laura-herrera) [reply aws/vpc/main.tf:2]: actually, most regions in Europe have 3 AZs, some of them have two

File: aws/vpc/outputs.tf (hunk 1 above)
Comment (laura-herrera) [reply aws/vpc/outputs.tf:1]: We can output any of the Attributes of a resource. The list is in the registry, in the page of the resource itself, right at the end. Seems we can output the tags_all attribute, which in our case does contain the name of the vpc .. along with the rest of the tags as a map

Comment (patrick-merchant) [reply aws/vpc/outputs.tf:1]: Great this sounds useful! I'll output the tags in the EC2 and ALB modules too

**Review by patrick-merchant — APPROVED** [review]

**Pull request merged**